core/dnsserver/server_grpc.go

```diff
@@ -40,6 +40,9 @@ func NewServergRPC(addr string, group []*Config) (*ServergRPC, error) {
 		// Should we error if some configs *don't* have TLS?
 		tlsConfig = conf.TLSConfig
 	}
+	// http/2 is required when using gRPC. We need to specify it in next protos
+	// or the upgrade won't happen.
+	tlsConfig.NextProtos = []string{"h2"}
```

Comment:
~move to inside for loop.~  Actually that wouldn't help, IIUC an explicit nil check is needed.

Reply:
Right, the nil check was only in https server.
Do you think we should also require tls here like what is done in [https server](https://github.com/coredns/coredns/blob/master/core/dnsserver/server_https.go#L42) as this is basically also https?

Reply:
Perhaps, though that might introduce backward incompatibility if currently we permit an unencrypted gRPC listener.

I was thinking just a simple nil check before accessing `NextProtos`.  e.g.

```suggestion
	If (tlsConfig != nil) {
		tlsConfig.NextProtos = []string{"h2"}
	}
```

Reply:
gprc with `WithInsecure()` is a normal thing to do when testing/playing with grpc, so we not make that impossible.
The h2 upgrade also makes sense in normal TLS setup... I think. It's not specified for DoT, but IIRC http/2 is negotiated, so it shouldn't harm either